python/test/light_test.sdf

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" ?>
+<sdf version="1.6">
+  <world name="world_test">
+
+    <light type="point" name="light_test">
```

Comment:
The `<direction>` and `<spot>` parameters do not apply when the light type is "point" (see http://sdformat.org/spec?ver=1.10&elem=light).

Reply:
Oh I see, we have `point`, `spot` and `directional`. That makes a lot more sense. That should solve some of my problems, but, I think the setters issue remains the same.

Reply:
I made the required changes in https://github.com/gazebosim/gz-sim/pull/2043/commits/4e72d74d66f78bfb3bfa0145198d1dea9033c0d9. However, the behavior is the same as I expected, solved some issues but not a single `set` method is working.